### ISSUE TITLE
docs: 1D eigenvalue & eigenfunction visualization

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/spectraldiffx#47]
## Summary

Add an interactive example visualizing the 1D eigenfunctions and eigenvalues for all transform types (DST-I/II/III/IV, DCT-I/II/III/IV, FFT). This builds foundational understanding of **why** each transform diagonalizes a specific discrete Laplacian.

## Motivation

The eigenvalue formulas look similar but produce qualitatively different basis functions depending on grid placement and boundary conditions. Seeing the eigenfunctions side-by-side makes the connection between transform type → BC type → grid type immediately obvious.

## File Locations

| File | Purpose |
|------|---------|
| `docs/examples/eigenfunction_gallery.md` | Rendered docs page |
| `notebooks/eigenfunction_gallery.py` | Jupytext percent-format notebook |
| `mkdocs.yml` | Add nav entry under Examples |

### Section 1: Same-BC Eigenfunctions (DST-I, DST-II, DCT-I, DCT-II)

Show the first 4 eigenfunctions (k=0..3) for each transform on N=16 points.

ASCII sketch of the key difference:

```
DST-I (Dirichlet, regular)           DST-II (Dirichlet, staggered)

  ψ=0 ·                    · ψ=0     ψ=0 |                    | ψ=0
       ·  o              o  ·         (wall)  x              x  (wall)
       ·    o          o    ·              x    x          x
       ·      o      o      ·            x      x      x
       ·        o  o         ·          x        x  x
       ·─────────────────────·       |─────────────────────|
       0  1  2  3  4  5  6  7       0  1  2  3  4  5  6  7

  o = grid point ON vertex            x = grid point AT cell centre
  ψ vanishes at grid points           ψ vanishes BETWEEN grid points
    0 and N+1 (implicit)                at walls half-spacing outside
```

```
DCT-I (Neumann, regular)            DCT-II (Neumann, staggered)

  ∂ψ/∂n=0                            ∂ψ/∂n=0
  |  o  o  o  o  o  o  o  o  |       |  x  x  x  x  x  x  x  x  |
  |                           |       |                           |
  boundary = first/last point         boundary = cell face
  N points including boundary         N points at cell centres
```

**matplotlib figure:** 4×2 grid of subplots:

```
┌──────────────────┬──────────────────┐
│ DST-I  k=0,1,2,3 │ DST-II k=0,1,2,3│
│ (Dir. regular)    │ (Dir. staggered) │
├──────────────────┼──────────────────┤
│ DCT-I  k=0,1,2,3 │ DCT-II k=0,1,2,3│
│ (Neu. regular)    │ (Neu. staggered) │
└──────────────────┴──────────────────┘
```

Each subplot shows 4 colored lines (one per eigenfunction), with grid points marked as dots. Boundary positions marked with vertical dashed lines.

### Section 2: Mixed-BC Eigenfunctions (DST-III, DST-IV, DCT-III, DCT-IV)

These arise when **different BCs** apply at each end of the domain:

```
DST-III: Dirichlet left, Neumann right     DCT-III: Neumann left, Dirichlet right
  ψ=0 |                        ∂ψ/∂n=0     ∂ψ/∂n=0                        | ψ=0
       x   x   x   x   x   x   |                x   x   x   x   x   x
       ·───────────────────────|            |───────────────────────·

DST-IV: Dirichlet both (half-integer)      DCT-IV: Neumann both (half-integer)
  ψ=0 |                        | ψ=0       ∂ψ/∂n=0 |                | ∂ψ/∂n=0
       x   x   x   x   x   x                   x   x   x   x   x   x
```

**matplotlib figure:** Same 4×2 layout for the mixed-BC types.

### Section 3: Eigenvalue Comparison

**matplotlib figure:** Single plot with all 9 eigenvalue curves (DST-I through IV, DCT-I through IV, FFT) on the same axes, N=32:

```
  λ_k
  0 ─── DCT-I ── DCT-II ── FFT ─────── k
  │      ╲         ╲        ╲
  │       ╲  mixed types (DST-III/IV, DCT-III/IV)
  │        ╲   all share same formula!
  │         ╲
  │          ╲ DST-II
  │           ╲
  │            ╲ DST-I
  -4/dx² ───────────────────────────── k=N-1
```

Key insight to highlight: **DST-III, DCT-III, DST-IV, DCT-IV all share the same eigenvalue formula** `λ_k = -4/dx² sin²(π(2k+1)/(4N))`. This should be called out in the narrative.

### Section 4: Eigenfunction Orthogonality

**matplotlib figure:** Heatmap of `Φᵀ Φ` (Gram matrix) for each transform type, showing orthogonality:

```
┌─────────┬─────────┬─────────┐
│ DST-I   │ DST-II  │ DCT-I   │
│ ΦᵀΦ ≈ I │ ΦᵀΦ ≈ I │ ΦᵀΦ ≈ I │
├─────────┼─────────┼─────────┤
│ DCT-II  │ FFT     │ DST-III │
│ ΦᵀΦ ≈ I │ ΦᵀΦ ≈ I │ ΦᵀΦ ≈ I │
└─────────┴─────────┴─────────┘
```

### Section 5: Stencil Verification

For each transform type, demonstrate that the eigenvalues are exact for the **discrete** Laplacian (tridiagonal matrix `[1, -2, 1]/dx²`):

```python
# Build tridiagonal Laplacian
L = (np.diag(np.ones(N-1), -1) - 2*np.diag(np.ones(N), 0) + np.diag(np.ones(N-1), 1)) / dx**2

# Verify: L @ phi_k = lambda_k * phi_k
```

**matplotlib figure:** Bar chart of `||L·φ_k - λ_k·φ_k||` for each k, showing machine-precision residuals.

## Code Structure

```python
# %% [markdown]
# # Eigenfunction & Eigenvalue Gallery

# %% imports
from spectraldiffx import (
    dst1_eigenvalues, dst2_eigenvalues, dst3_eigenvalues, dst4_eigenvalues,
    dct1_eigenvalues, dct2_eigenvalues, dct3_eigenvalues, dct4_eigenvalues,
    fft_eigenvalues,
    dst, dstn, idst, idstn, dct, dctn, idct, idctn,
)

# %% Section 1: Build eigenfunctions from transforms
def dst1_eigenfunctions(N):
    """Columns of the DST-I matrix."""
    return jnp.eye(N)  # apply dst type=1 to each column

# %% Section 2-5: plots
```

## Acceptance Criteria

- [ ] All 9 transform types visualized
- [ ] Narrative explains the connection: transform type → BC type → grid placement
- [ ] Mixed-BC types explicitly noted as sharing the same eigenvalue formula
- [ ] Stencil verification confirms machine-precision eigenvalue accuracy
- [ ] Notebook runs cleanly with `uv run jupyter execute`

## Labels

`documentation`, `examples`
